.github/workflows/lint.yml

```diff
@@ -0,0 +1,33 @@
+
+name: Lint and Format
+
+on: [push, pull_request]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        name: Clone repo
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '14'
+        name: Set up Node
+      - run: npm ci
+        name: Install dependencies
+      - run: npm run lint:fix
+        name: Run Linter
+      - name: Get changed files
+        id: changes
+        # Set outputs using the command.
+        run: |
+          echo "::set-output name=all::$(git diff --name-only --diff-filter=ACMRT ${{ github.event.pull_request.base.sha }} ${{ github.sha }} | xargs)"
+          echo "::set-output name=ts::$(git diff --name-only --diff-filter=ACMRT ${{ github.event.pull_request.base.sha }} ${{ github.sha }} | grep .ts$ | xargs)"
+      - if: ${{ steps.changes.outputs.all }}
+        name: Commit changes
+        run: |
+          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git config --global user.name "GitHub Actions"
+          git add .
+          git commit -m "style: lint and format"
+          git push
```

Comment:
We don't want to have to fight the linter workflow in the event that ESLint/Prettier is enforcing a change that we do not want. I'd rather we not have the workflow automatically commit fixes.